scm-webapp/src/main/java/sonia/scm/plugin/PluginCenterDtoMapper.java

```diff
@@ -28,19 +28,39 @@
 import org.mapstruct.factory.Mappers;
 
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 @Mapper
 public abstract class PluginCenterDtoMapper {
 
+  PluginCenterDtoMapper() {}
+
   static final PluginCenterDtoMapper INSTANCE = Mappers.getMapper(PluginCenterDtoMapper.class);
 
   abstract PluginInformation map(PluginCenterDto.Plugin plugin);
 
   abstract PluginCondition map(PluginCenterDto.Condition condition);
 
-  Set<AvailablePlugin> map(PluginCenterDto pluginCenterDto) {
+  PluginCenterResult map(PluginCenterDto pluginCenterDto) {
     Set<AvailablePlugin> plugins = new HashSet<>();
+    Set<PluginSet> pluginSets = new HashSet<>();
+    for (PluginCenterDto.PluginSet pluginSet : pluginCenterDto.getEmbedded().getPluginSets()) {
+      pluginSets.add(new PluginSet(
+        pluginSet.getId(),
+        pluginSet.getSequence(),
+        pluginSet.getPlugins(),
+        pluginSet.getDescriptions()
+          .entrySet()
+          .stream()
+          .collect(Collectors.toMap(
+            Map.Entry::getKey,
+            e -> new PluginSet.Description(e.getValue().getName(), e.getValue().getFeatures())
+          )),
+        pluginSet.getImages()
```

Comment:
To avoid missing mappings for future attributes this should be done by an own mapping with mapstruct.
```java
  abstract PluginSet map(PluginCenterDto.PluginSet set);
```